### PR TITLE
Support for additional Appoptics Host Agent Plugins (AO-9502)

### PR DESCRIPTION
**Pull request by gortyy**
https://swicloud.atlassian.net/browse/AO-9502

Host Agent in container now has support for:
*  Apache
*  Docker
*  Elasticsearch
*  Kubernetes
*  Mesos
*  Mongodb
*  Mysql
*  Rabbitmq
*  Zookeeper
*  Psutil

Each plugin can be enable by setting `APPOPTICS_ENABLE_<plugin_name>` to `true`.
In case of Psutil plugin, the flag is `APPOPTICS_DISABLE_HOSTAGENT` and to turn the plugin, it should be set to `false`. By default only Docker plugin is turned on.

### CODE REVIEW

File: appoptics-agent-deployment.yaml [hunk 1]
```diff
@@ -18,13 +18,29 @@ spec:
...
               value: 'APPOPTICS_TOKEN'
-            - name: APPOPTICS_ENABLE_KUBERNETES
+            - name: APPOPTICS_ENABLE_APACHE
+              value: 'false'
+            - name: APPOPTICS_ENABLE_DOCKER
               value: 'true'
+            - name: APPOPTICS_ENABLE_ELASTICSEARCH
+              value: 'false'
+            - name: APPOPTICS_ENABLE_KUBERNETES
+              value: 'false'
```
Comment (cmrust) [inline appoptics-agent-deployment.yaml]: This deployment manifest is specifically for the Kubernetes plugin. The README.md outlines instructions for using this file in a specific manner, where only the Kubernetes plugin is enabled. Changing all of these default values breaks that.

If we want to offer an example manifest with the available ENV vars exposed in "default" form we should create a separate file here in the project.

Comment (cmrust) [reply appoptics-agent-deployment.yaml]: APPOPTICS_ENABLE_KUBERNETES and APPOPTICS_DISABLE_HOSTAGENT should be true and everything else should be false.

**Comment by cmrust** [comment]
@gortyy Great changes. Can you respond to my comments above? I will approve after.

Also, there is no CICD for this image creation. Do you have credentials to release this new version to Docker Hub? If not, let me know and I will assist.

File: appoptics-agent-deployment.yaml (hunk 1 above)
Comment (gortyy) [reply appoptics-agent-deployment.yaml]: @cmrust I've updated  vars, thanks for the catch

**Comment by cce** [comment]
Even if they enable APPOPTICS_ENABLE_ELASTICSEARCH, or RabbitMQ, Apache, or MongoDB, etc, they will have to modify the config file somehow to set the hostname/port of the ES, RabbitMQ, Apache, MongoDB server, right? It would be nice to have something similar to `MYSQL_HOST` for that, or at least something in the documentation mentioning this.. or maybe I missed it.

**Comment by cmrust** [comment]
@cce I believe most people monitoring specific services will take the Sidecar approach detailed in the README. When you run the agent as a sidecar (second container in pod) they share localhost. 

My impression is that some of these plugins (though i'm not familiar with all of them) can find the service without an address or credentials if they can see it at the default port on localhost. I believe this is how the Zookeeper example from the README works, for example.

That said, we do have the single-pod deployment specifically for the Kubernetes plugin that we suggest users deploy. I see no reason we couldn't turn on these plugins individually there and pass them whatever they need for finding their services, to centralize and lower the amount of agents a user has to distribute.

Not trying to discourage improving the docs or adding more ENV vars (it is probably necessary on some of these), just adding details.

**Comment by tomekkolo** [comment]
@cce I suppose that the intention is to provide configuration by mapping folder /opt/appoptics/etc/plugins.d or tasks.d. then conf/appoptics-init.sh will just enable required config files based on env vars. @gortyy this could be actually added to README. That if customer wants to have custom config (i.e. not using localhost), then he should map/copy config files. 

Adding more env variables to cover all configs can result in very long list.

**Comment by gortyy** [comment]
Users can either set `APPOPTICS_ENABLE_<plugin>` and use default plugin config or mount config files directly to `/opt/appoptics/plugins.d/` without setting the env vars. I'll update README

**Comment by gortyy** [comment]
I've added configuration instructions to README

**Comment by cce** [comment]
Cool thanks @cmrust @gortyy OK yes some of the configs do default to localhost — checking them this is true for Apache, Elasticsearch, Mesos, MongoDB, MySQL, RabbitMQ, Zookeeper. Cool!

**Comment by leecalcote** [comment]
This one of our few differentiators in our Kubernetes story. It’s not that others doesn’t have it, but that everyone doesn’t.

- Lee

Sent from my mobile
________________________________

**Pull request merged**